app/controllers/api/v1/groups_controller.rb

```diff
@@ -0,0 +1,256 @@
+      # |:------------------|:----------------------------------------|
+      # | `article_names`   | The list of article names in the group  |
+      #
+      # Returns
+      # -------
+      # a map from article names to their readiness
+
+      def update
+        errors = update_article_groups(@group, params['article_names'])
+        if errors
+          render_errors(errors)
+          return
+        end
+
+        respond_with @project do |format|
+          format.json { render json: decorate_group(@group) }
+        end
+      end
+
+      def destroy
+        @group.destroy
+
+        respond_with @project do |format|
+          format.json { render json: { status: true } }
+        end
+      end
+
+      # private
+
+      # ===== START AUTHENTICATION/AUTHORIZATION/VALIDATION ============================================================
+      def authenticate_with_api_token!
+        unless Project.where(id: params[:project_id], api_token: params[:api_token]).exists?
+          render_error_with_message(t('controllers.api.v1.groups.invalid_api_token'), :unauthorized)
+        end
+      end
+
+      def find_project
+        @project = Project.find_by_id(params[:project_id])
+
+        unless @project
+          render_error_with_message(t('controllers.api.v1.groups.project_not_found'), :not_found)
+        end
+      end
+
+      def find_group
+        @group = Group.find_by_name(params[:name])
+
+        unless @group
+          render_error_with_message(t('controllers.api.v1.groups.group_not_found'), :not_found)
+        end
+      end
+
+      def api_request?
+        params[:api_token].present?
+      end
+
+      # ===== END AUTHENTICATION/AUTHORIZATION/VALIDATION ==============================================================
+
+      # ===== START DECORATORS =========================================================================================
+      def decorate_groups
+        @project.groups.map(&:name).sort
+      end
+
+      def decorate_group(group)
+        group.article_groups.includes(:article).order(:index_in_group).map do |article_group|
+          {
+              name: article_group.article.name,
+              ready: article_group.article.ready?
+          }
+        end
+      end
+      # ===== END DECORATORS ===========================================================================================
+
+      def group_exists?(name)
+        @project.groups.where(name: name).count > 0
+      end
+
+      def render_error_with_message(message, status)
+        render_errors([{ message: message }], status)
+      end
+
+      def render_errors(errors, status=:bad_request)
+        respond_with(nil) do |format|
+          format.json { render json: { error: { errors: errors } }, status: status }
+        end
+      end
+
+      def update_article_groups(group, article_names)
```

Comment:
Are the existent relations being deleted before the new ones are created?
I see it is tested in the spec file, but I don't see in the code where it is done.

Reply:
The existing relationships are stored in article_groups and replaced with the new ones, which is build from the input articles.